src/ensembl/src/content/app/custom-download/components/paste-or-upload/PasteOrUpload.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useEffect } from 'react';
+import Input from 'src/shared/input/Input';
+import ImageButton from 'src/shared/image-button/ImageButton';
+import { ReactComponent as RemoveIcon } from 'static/img/shared/clear.svg';
+import styles from './PasteOrUpload.scss';
+import noop from 'lodash/noop';
+import get from 'lodash/get';
+
+type PasteOrUploadProps = {
+  value: string | null;
+  placeholder?: string;
+  onChange: (value: string) => void;
+  onRemove: () => void;
+};
+
+const PasteOrUpload = (props: PasteOrUploadProps) => {
+  const [shouldShowInput, showInput] = useState(props.value !== null);
+  const [value, setValue] = useState<string | null>(null);
+
+  useEffect(() => {
+    setValue(props.value);
+  }, [props.value]);
+
+  const [shouldShowFileUpload, showFileUpload] = useState(false);
+  const onChangeHandler = (value: string) => {
+    props.onChange(value);
```

Comment:
So if every change event of the input calls onChange on the props, and props, when changed, update component’s state, then what’s the point of having component’s state?

Reply:
It's not necessary. I've removed it.